exporter/exporterhelper/internal/queue_sender.go

```diff
@@ -17,9 +17,8 @@ import (
 
 // NewDefaultQueueConfig returns the default config for queuebatch.Config.
 // By default, the queue stores 1000 requests of telemetry and is non-blocking when full.
-func NewDefaultQueueConfig() queuebatch.Config {
-	return queuebatch.Config{
-		Enabled:      true,
+func NewDefaultQueueConfig() configoptional.Optional[queuebatch.Config] {
+	return configoptional.Some(queuebatch.Config{
```

Comment:
Following up from an offline discussion, would it be a major inconvenience if we just returned the config directly here and required consumers to wrap it themselves? I feel like it would make tests easier and would make the function slightly less opinionated.

Reply:
I agree!

I still feel something's not perfect. We're recommending that users call configoptional.Default(...) or configoptional.Some(...) in order to disable or enable a feature by default. It sort of suggests adding a function named `configoptional.Enabled(...)` (for enabled with defaults) and `configoptional.Disabled(...)` (for disabled with defaults).

Reply:
I see your concern. I think this partially sources from the fact that we have the `enabled` field in configoptional now, as before it was just a way to think about the presence of a value at the time the function is called. The best I can think of off the top of my head is aliasing each function to `EnabledByDefault` and `DisabledByDefault` or similar, but I'd want to give it more thought before committing. If we go this route, would want the names to be very explicit since I think `Enabled` and `Disabled` are also a little ambiguous.